ons_alpha/core/blocks/panels.py

```diff
@@ -22,12 +26,34 @@ class Meta:
         template = "templates/components/streamfield/panel_block.html"
 
 
+class PreviousVersionBlock(blocks.IntegerBlock):
+    pass
+
+
 class CorrectionBlock(blocks.StructBlock):
     when = blocks.DateTimeBlock()
     text = blocks.RichTextBlock(features=settings.RICH_TEXT_BASIC)
-    previous_version = ...
+    previous_version = PreviousVersionBlock(required=False)
+
+    class Meta:
+        template = "templates/components/streamfield/corrections_block.html"
+
+
+class PreviousVersionBlockAdapter(FieldBlockAdapter):
+    js_constructor = "ons_alpha.core.blocks.panels.PreviousVersionBlock"
+
+    @cached_property
+    def media(self):
+        structblock_media = super().media
+        return Media(js=structblock_media._js + ["js/previous-version-block.js"], css=structblock_media._css)
```

Comment:
You can achieve the same via a custom `form_template` and passing the extra context in `get_form_context` on `PreviousVersionBlock`. See https://docs.wagtail.org/en/stable/advanced_topics/customisation/streamfield_blocks.html#custom-editing-interfaces-for-structblock

Reply:
You can, but also you can't. As part of rendering the template, I specifically need the value, which isn't present when `form_template` is rendered, as all that is handled client-side. So sadly, this JS is needed :disappointed: 

If we just wanted to hide the `previous_version` field, this would work. But to show the buttons we need its current value.

Reply:
yeah, that is annoying. let's leave it as it is